org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -100,6 +101,8 @@ protected IStatus run(IProgressMonitor monitor) {
 
 	private IPreferencesChangeListener preferenceChangeListener = null;
 
+	private IPreferencesChangeListener gradlePreferenceChangeListener = new GradlePreferenceChangeListener();
```

Comment:
again, we should ensure StandardProjectsManager is build-tool agnostic. We shouldn't reference anything specific to Gradle (or any other tool) in this class.

Reply:
We already have such reference. See https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java#L162